File: Linguagem C++/Matrizes/SomaLinhas.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    int m,n;

    cout << "Qual a quantidade de linhas da matriz? ";
    cin >> m;
    cout << "Qual a quantidade de colunas da matriz? ";
    cin >> n;

    double mat[m][n];

    /// Lendo a matriz
    for (int i=0;i<=m-1;i++){
        cout << "Digite os elementos da " << i+1 << "a. linha: " << endl;
        for (int j=0;j<=n-1;j++){
            cin >> mat[i][j];
        };
    };

    /// Somando as linhas e gerando o vetor
    double vet[n];

    cout << "VETOR GERADO:" << endl;
    for (int i=0;i<=m-1;i++){
        vet[i] = 0;
        for (int j=0;j<=n-1;j++){
            vet[i] = vet[i] + mat[i][j];
        };
        cout << fixed << setprecision(1);
        cout << vet[i] << endl;
    };

    return 0;
}

File: Linguagem Java/Exercicios/SomaLinhas.java
import java.util.Locale;
import java.util.Scanner;

public class SomaLinhas {

	public static void main(String[] args) {
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		int m,n;
		double soma;
		
		System.out.print("Qual a quantidade de linhas da matriz? ");
		m = sc.nextInt();
		System.out.print("Qual a quantidade de colunas da matriz? ");
		n = sc.nextInt();
		
		double mat[][] = new double[m][n];
		double vet[] = new double [m];
		
		// Lendo a matriz
		for (int i=0;i<m;i++) {
			System.out.println("Digite os elementos da "+(i+1)+"a linha:");
			for (int j=0;j<n;j++) {
				mat[i][j] = sc.nextDouble();				
			};			
		};
		
		// Somando as linhas
		for (int i=0;i<m;i++) {
			soma = 0;
			for (int j=0;j<n;j++) {
				soma = soma + mat[i][j];				
			};
			vet[i] = soma;
		};
		
		//Imprimindo o vetor
		System.out.println("VETOR GERADO:");
		for (int i=0;i<n;i++) {
			System.out.println(vet[i]);			
		};
		
		sc.close();

	}

}
